x/mint/keeper/genesis_test.go

```diff
@@ -4,26 +4,183 @@ import (
+		isDeveloperModuleAccountCreated bool
+
+		expectPanic             bool
+		expectedEpochProvisions sdk.Dec
+		// Deltas represent by how much a certain paramets
+		// has changed after calling InitGenesis()
+		expectedSupplyOffsetDelta           sdk.Int
+		expectedSupplyWithOffsetDelta       sdk.Int
+		expectedDeveloperVestingAmountDelta sdk.Int
+		expectedHalvenStartedEpoch          int64
+	}{
+		"default genesis - developer module account is not created prior to InitGenesis() - created during the call": {
+			mintGenesis: types.DefaultGenesisState(),
+			mintDenom:   sdk.DefaultBondDenom,
+
+			expectedEpochProvisions:             types.DefaultGenesisState().Params.GenesisEpochProvisions,
+			expectedSupplyOffsetDelta:           sdk.NewInt(keeper.DeveloperVestingAmount).Neg(),
+			expectedSupplyWithOffsetDelta:       sdk.ZeroInt(),
+			expectedDeveloperVestingAmountDelta: sdk.NewInt(keeper.DeveloperVestingAmount),
+		},
+		"default genesis - developer module account is created prior to InitGenesis() - not created during the call": {
+			mintGenesis:                     types.DefaultGenesisState(),
+			mintDenom:                       sdk.DefaultBondDenom,
+			isDeveloperModuleAccountCreated: true,
+
+			expectedEpochProvisions:             types.DefaultGenesisState().Params.GenesisEpochProvisions,
+			expectedSupplyOffsetDelta:           sdk.ZeroInt(),
+			expectedSupplyWithOffsetDelta:       sdk.ZeroInt(),
+			expectedDeveloperVestingAmountDelta: sdk.ZeroInt(),
+		},
+		"custom genesis": {
+			mintGenesis: customGenesis,
+			mintDenom:   "uosmo",
+
+			expectedEpochProvisions:             sdk.NewDec(200),
+			expectedSupplyOffsetDelta:           sdk.NewInt(keeper.DeveloperVestingAmount).Neg(),
+			expectedSupplyWithOffsetDelta:       sdk.ZeroInt(),
+			expectedDeveloperVestingAmountDelta: sdk.NewInt(keeper.DeveloperVestingAmount),
+			expectedHalvenStartedEpoch:          3,
+		},
+		"nil genesis state - panic": {
+			mintDenom:   sdk.DefaultBondDenom,
+			expectPanic: true,
+		},
+	}
+
+	for name, tc := range testCases {
+		suite.Run(name, func() {
+			// Setup.
+			suite.setupDeveloperVestingModuleAccountTest(tc.ctxHeight, tc.isDeveloperModuleAccountCreated)
+			ctx := suite.Ctx
+			accountKeeper := suite.App.AccountKeeper
+			bankKeeper := suite.App.BankKeeper
+			mintKeeper := suite.App.MintKeeper
+
+			developerAccount := accountKeeper.GetModuleAddress(types.DeveloperVestingModuleAcctName)
+
+			originalSupplyOffset := bankKeeper.GetSupplyOffset(ctx, tc.mintDenom)
+			originalSupplyWithOffset := bankKeeper.GetSupplyWithOffset(ctx, tc.mintDenom)
+			originalVestingCoins := bankKeeper.GetBalance(ctx, developerAccount, tc.mintDenom)
+
+			// Test.
+			if tc.expectPanic {
+				suite.Panics(func() {
+					mintKeeper.InitGenesis(ctx, accountKeeper, bankKeeper, tc.mintGenesis)
+				})
+				return
+			}
+
+			suite.NotPanics(func() {
+				mintKeeper.InitGenesis(ctx, accountKeeper, bankKeeper, tc.mintGenesis)
+			})
```

Comment:
We should make a helper for this pattern. e.g. `osmoutils.ConditionalPanics(suite, expPanic, func)`

Lets file an issue for this?

Reply:
https://github.com/osmosis-labs/osmosis/issues/1946